crates/rbuilder/src/primitives/mod.rs

```diff
@@ -742,17 +760,17 @@ impl TransactionSignedEcRecoveredWithBlobs {
         T: TransactionOrdering<Transaction = <V as TransactionValidator>::Transaction>,
         S: BlobStore,
     {
-        let blob_sidecar = pool
-            .get_blob(*tx.inner().hash())?
-            .and_then(|b| b.as_eip4844().cloned());
+        let mut blobs = pool.get_all_blobs(vec![*tx.inner().hash()])?;
```

Comment:
`pool.get_blob` wasn't able to retrive the blob in my testing. using `pool.get_all_blobs` worked in this case.